packages/terra-responsive-element/src/ResponsiveElement.jsx

```diff
@@ -61,7 +61,7 @@ class ResponsiveElement extends React.Component {
       this.resizeObserver = new ResizeObserver((entries) => {
         this.animationFrameID = window.requestAnimationFrame(() => {
           this.animationFrameID = null;
-          this.handleResize(entries[0].contentRect.width);
+          this.handleResize(entries[0].target.getBoundingClientRect().width);
```

Comment:
I'm not sure if getting the parentNode of the target is what we want. The target should already be the container for the responsive elements, and that container is what is what is (conditionally) rendering the scrollbar. If we go a level higher, we're in uncharted waters.

I wouldn't have expected the width from getBoundingClientRect to differ from the offsetWidth of the same element.

Reply:
@StephenEsser In above code example there is outer div for responsive element hence the target value is set as <div> and the target.parentnode is set as parentcontainer. so we are getting the issue of repetitive resizing using my initial code. if we add one more outerdiv to the current div the issue still exists with newly suggested code as the target.parentnode will be the outermost div and target will be innerdiv.

Observerresize.target will always hold the immediate parent control of responsiveelement. if the div elements are removed **entries[0].target.getBoundingClientRect().width** works fine as the target is always set to immediate parent of the responsive element. without any outer div elements the target.parentnode value will be set to parent container of responsive element parent container.

i even tried creating sample react app consuming the responsive element without any blank outer div elements. in this case the target value is set to root div element where as target.parentnode is html body.

so my question here is will there be a real time scenario where blank div be there around responsive element ?

Reply:
@bjankord , @StephenEsser request you guys to review this PR. please let me know if i need to test anything else.

Reply:
Is there a scenario where the code change above prevents an infinite resize?

@tbiethman @bjankord and I spoke offline and there is not currently a known solution that fixes the infinite resizing in all cases and it's possible there isn't one. The infinite resizing is caused by the available space fluctuating by a set amount of pixels that pushes each resize into the other breakpoint threshold.  

Unless the solution above fixes any of the infinite resizing use-cases I would vote to close this PR and replace it with a PR that documents some of the caveats and recommended use cases of the ResponsiveElement. Specifically the caveat that if any ancestor's overflow can be affected by the contents within the ResponsiveElement it can result in an infinite resize. 

@tbiethman  @bjankord Do you guys have any thoughts on this?

Reply:
@StephenEsser Regarding your question "Is there a scenario where the code change above prevents an infinite resize?"

Yes, the above code change prevents the infinite resize during the scrollbar display while shifting to next immediate breakpoint of responsive element. the scrollbar is due to parent control which has the overflow property set to 'Auto'. 

the previous code **'entries[0].contentRect.width'** used to not take scrollbar width into account which would make the breakpoint of responsive element shift to one level below. while it should actually not shift the breakpoints. hence using parent control width which includes scrollbar width also.

Reply:
I believe the width returned from `entries[0].target.getBoundingClientRect().width` also includes the padding and borders. The ResponsiveElement is currently implemented to shift between breakpoints based on the available space for the content. There are a few notes about this on the [ReadMe](https://github.com/cerner/terra-core/blob/master/packages/terra-responsive-element/docs/README.md):

> The ResponsiveElement observes the available width. Meaning padding, borders, and margins set on the parent are not included in the calculation. Modifying box-sizing will impact how these properties are used in an elements width calculation. For more details see: https://css-tricks.com/almanac/properties/b/box-sizing/

I think the ResponsiveElement should continue to honor the available space for calculations and not the raw width of the parent including paddings / borders. 

Here's a short example.

```
<div id="parent" style="padding: 20px; border: 5px solid black; width: 100px;">
  <div>ResponsiveElement</div>
</div>
```

In this example the bounding rect width would report 150px of space, but there is actually only 100px of available space for the ResponsiveElement.

Reply:
Had a chance to look over this PR and the related issue in more detail. @StephenEsser created a [branch with a reduced test case](https://github.com/cerner/terra-core/tree/InfiniteResizeExamples) which was helpful to reason about the issue. I'm okay with updating the responsive element behavior to report the **raw width of the parent including paddings / borders** instead of the **available space** as it was prior, but I would consider this to be a non-passive change.

I'd like to see some documentation updates noting the behavior change as the docs currently mention reporting the available space.

 e.g. Before this PR, if a div has `width: 120px` and `padding: 10px`, it would report `100px` of available space. With the change in this PR, the responsive element would now report `120px` of available space even though there is `10px` padding on the sides. I think this concern can be mitigated with some documentation.

In regards to this being a non-passive change, I'd like to hold off on merging this PR until we are ready to incorporate [the other non-passive changes we have planned in this issue](https://github.com/cerner/terra-ui/issues/117). That work will likely happen sometime in Q1.

Reply:
Updated the documentation.